commit-message-hook.sh

```diff
@@ -28,3 +28,23 @@ fi
     echo "Branch name is not named properly, please see wiki for formating: https://github.com/space-concordia-robotics/robotics-prototype/wiki/Work-Flow"
     exit 1
 fi
+#Checks to see if branch name includes any upper case letters. If it does, it exits and tells user to check wiki
+if [[ $BRANCH_NAME =~ [\$A-Z] ]];then
+# If any upper case letters are in the branch name,
+  echo "Branch name is not named properly, please see wiki for formating: https://github.com/space-concordia-robotics/robotics-prototype/wiki/Work-Flow"
+  exit 1
+fi
+
+TRIM_NUM=$(echo $BRANCH_NAME | sed -e 's/[0-9]//g' )
+#Checks to see if branch name includes a hypen seperator before the issue number. If not, it exits and tells user to check wiki
+if [[ "${TRIM_NUM: -1}" != *-* ]];then
+# If a hyphen is not found,
+  echo "Branch name is not named properly, please see wiki for formating: https://github.com/space-concordia-robotics/robotics-prototype/wiki/Work-Flow"
+  exit 1
+fi
+#Checks to see if branch name includes various other symbols. If so, it exits and tells user to check wiki
+if [[ $BRANCH_NAME == *['!'@\$%^\&*()_+]* ]];then
```

Comment:
carets `^`, asterisks `*` are not valid in git branch names. I would suggest to exclude them from here since they are already guaranteed not to be in a valid branch name.

Also you should include `]` closed bracket since by default it is allowed (although `[` by default is not, so no need to include it).

Make sure to check here on which symbols are allowed by default:
https://mirrors.edge.kernel.org/pub/software/scm/git/docs/git-check-ref-format.html